packages/exp/src/commands/upload/AndroidUploader.js

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs';
+
+import prompt from '../../prompt';
+import log from '../../log';
+
+import BaseUploader from './BaseUploader';
+import { printError, spawnAndCollectJSONOutputAsync } from './utils';
+
+export default class AndroidUploader extends BaseUploader {
+  constructor(projectDir, options) {
+    super(projectDir, options, 'android', 'Android', 'apk');
+  }
+
+  ensurePlatformOptionsAreCorrect() {
+    const { key } = this.options;
+    if (key && !fs.existsSync(key)) {
+      throw new Error(`No such file: ${key}`);
+    }
+  }
+
+  ensureConfigDataIsCorrect(configData) {
+    const { android } = configData;
+    if (!android || !android.package) {
+      throw new Error(`Must specify a package in order to upload apk file.`);
+    }
+  }
+
+  getPlatformData() {
+    if (!this.options.key) {
+      log('You can specify json file ID using --key option');
+      return prompt({
```

Comment:
In this case, you're not validating whether the path supplied by a user is correct.

Reply:
Fastlane will check if this file is correct